roles/dns-server-detect/tasks/main.yml

```diff
@@ -28,8 +29,9 @@
     public_dns_server: "{{ external_nsupdate_keys['public']['server'] }}"
   when:
     - external_nsupdate_keys['public'] is defined
+    - hostvars[groups['dns'][0]] is undefined
```

Comment:
yeah, I suggested that the dns group override external nsupdate keys' server entries. This has to be done if we want to support pre-provisioned by a Heat stack DNS servers for future runs. I look forward for better solutions for that use case...